examples/ProgressLinear.isAnimated.jsx

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import { ProgressLinear } from '@itwin/itwinui-react';
+
+export default () => {
+  const [value, setValue] = React.useState(0);
+
+  React.useEffect(() => {
+    setTimeout(() => {
+      const newValue = value + 20;
+      setValue(newValue > 100 ? 0 : newValue);
+    }, 3000);
```

Comment:
This demo can be improved in a few ways.

1. Shorten the timeout from 3s to 2s (or 1.5s) so that the animation is more obvious.
2. Don't loop it; it can be a bit annoying when you're just trying to read the docs. Instead, display a completion message and set the `status`.
3. Provide a "Reset" button for restarting the timeout, in case the user missed the animation.
4. In the MDX, change the client directive from `client:load` to `client:visible`, so that it gets triggered when the demo comes into view. (See [docs](https://docs.astro.build/en/reference/directives-reference/#clientvisible))

Reply:
I have added a message and status as well as a reset button.

Reply:
It doesn't seem to work in the [deploy preview](https://deploy-preview-2271--itwinui-previews.netlify.app/docs/progressindicator/). It's probably because of `useEffect` running too often (since it clears the timeout every time the `value` changes).

To fix this, you can create a variable (outside the `useEffect`): `const isDone = value === 100;` and use that variable (instead of `value`) in the `useEffect`.

Reply:
I have used `isDone` instead of `value` but for some reason it didn't work. I tried switching back to `client:load` and the demo ran so I'm not sure if that's the problem.

Reply:
You are still using `value` in the dependency array of `useEffect`, it needs to be just `[isDone]`.

Reply:
The other issue I noticed is that `setTimeout` should be `setInterval` so that it continuously re-runs until it's cleared.

If it still does not work, it might be a bug in `client:visible`. Let's use `client:load` for now and investigate later.

Reply:
I have switched to `setInterval` and added the `client:visible` issue as an after-PR todo item.

Reply:
Some more suggestions:

- We don't need to use a ref for the timeout; it can be a local variable inside the Effect.
- We should turn off `isAnimated` when value reaches 100, to avoid unnecessary/artificially delayed value. (It looks strange when the green progress bar animates).
- We should change the `key` when resetting the process, to force a remount and avoid unnecessary animation when the value goes from 100 to 0.

<details><summary>Full code</summary>

```jsx
export default () => {
  const [value, setValue] = React.useState(0);
  const isDone = value === 100;

  React.useEffect(() => {
    let interval;

    if (!isDone) {
      interval = setInterval(
        () => setValue((prevValue) => prevValue + 20),
        1500,
      );
    }

    return () => clearTimeout(interval);
  }, [isDone]);

  return (
    <div className='demo-container'>
      <ProgressLinear
        value={value}
        isAnimated={!isDone}
        labels={!isDone ? ['Loading...', `${value}%`] : ['Upload succeeded.']}
        status={isDone ? 'positive' : undefined}
        key={isDone.toString()}
      />
      <Button onClick={() => setValue(0)}>Reset process</Button>
    </div>
  );
};
```

</details>

Reply:
I have done as suggested.

Reply:
`clearTimeout` → `clearInterval` (typo in my original suggestion, sorry)

Reply:
Oh I just noticed that when you commented too. I have fixed it!